[MLOB-3136] Mention structured output support for azure BYOK eval (#30258)
DataDog/documentation: content/en/llm_observability/evaluations/ootb_evaluations.md

<div class="alert alert-info">Azure OpenAI is not supported for HIPAA organizations with a Business Associate Agreement (BAA) with Datadog.</div>

> Connect your Azure OpenAI account to LLM Observability with your OpenAI API key. We strongly recommend using the `GPT-4o mini` model for evaluations. The selected model version has to support [structured output][8].

Reviewer: ```suggestion
Connect your Azure OpenAI account to LLM Observability with your OpenAI API key. Datadog strongly recommends using the `GPT-4o mini` model for evaluations. The selected model version must support [structured output][8].
```
